packages/components/src/draggable/index.js

```diff
@@ -61,6 +82,29 @@ export default function Draggable( {
 		}
 	}
 
+	/**
+	 * @param {import('react').DragEvent} e
+	 * @param {DragState} dragState
+	 */
+	function over( e, dragState ) {
+		const nextX = dragState.x + e.clientX - dragState.cursorLeft;
+		const nextY = dragState.y + e.clientY - dragState.cursorTop;
+		if ( dragState.scale === 1 ) {
+			dragState.cloneWrapper.style.transform = `translate( ${ nextX }px, ${ nextY }px )`;
+		} else {
+			dragState.cloneWrapper.style.transform = `translate( ${ nextX }px, ${ nextY }px ) scale( ${ dragState.scale } )`;
+		}
+
+		// Update cursor coordinates.
+		dragState.cursorLeft = e.clientX;
+		dragState.cursorTop = e.clientY;
+		dragState.x = nextX;
+		dragState.y = nextY;
```

Comment:
I wonder if it might be confusing updating the object properties in this way rather than through a closure. 🤔 

Maybe here it could call an `updateDragState` function that makes the change to a `ref` or something.

Reply:
I tried removing the dragState object in https://github.com/WordPress/gutenberg/pull/33395/commits/7e1bda9c910355923f16d48e2933a2dcc5485f16 and removing the extra function.

Closures are pretty common in animation/drawing handlers (since we need to keep track of what happened in the last step vs the current step), but we could try to tidy up function signatures in a larger refactor. I'd maybe move that to a seperate PR though.